lib/sales_analyst.rb

```diff
@@ -19,15 +19,13 @@ def mean_calculation_merchant(merchant)
   end
 
   def average_items_per_merchant_standard_deviation
-    mean = merchants.all.map do |merchant|
-      mean_calculation_merchant(merchant)
-    end
-    calculate_stdev(mean)
+    calculate_stdev(merchant_mean)
   end
 
   def merchants_with_high_item_count
+    merchant_stdev = average_items_per_merchant_standard_deviation
     all_merchants.values.find_all do |merchant|
-      merchant.items.count > (average_items_per_merchant_standard_deviation * 2)
+      merchant.items.count > (merchant_stdev * 2)
```

Comment:
See this? We were running the STDev every time that we pulled one object out.

Reply:
Right! This was what I was getting at but wasn't sure how to implement. Killer.